osg_configure/configure_modules/sge.py

```diff
@@ -111,9 +110,8 @@ def check_attributes(self, attributes):
                      level=logging.ERROR)
 
         settings_file = os.path.join(self.options['sge_root'].value,
-                                     self.options['sge_cell'].value,
-                                     'common',
-                                     'settings.sh')
+                                     self.options['sge_cell'].value.lstrip("/"),
```

Comment:
was a leading `/` for `sge_cell` a bug in python2 then also?

Reply:
You never want to have a leading `/` in the 2nd..Nth arguments of `os.path.join()` because they replace the previous path instead of appending to it. This is true in both Python 2 and 3.